src/webpack-angular-types-plugin/type-extraction.ts

```diff
@@ -57,15 +70,123 @@ function getJsDocs(node: JSDocableNode): string {
     );
 }
 
+function truncateImportPart(typeStr: string): string {
+    const regex = /import\(.+\)\./g;
+    return typeStr.replace(regex, "");
+}
+
+function expandTypeToString(type: Type, level = 0): string {
+    if (type.isUnion() || type.isIntersection()) {
+        const types = type.isUnion()
+            ? type.getUnionTypes()
+            : type.getIntersectionTypes();
+        const joinSymbol = type.isUnion() ? " | " : " & ";
```

Comment:
I'm not sure if we should move output-styling to the args-table-mapper instead, what do you think? Let's talk about this on Monday

I can imagine this would make things more complex here. Though, if the webpack plugin only passes a data structure that others can consume it might be easier to change the output to whatever the consumer wants. The more I think about it, the more it feels like the webpack plugin should be a separate package that provides angular types and others can do whatever they want with it. Nothing we need to do right now, of course

Reply:
I am not quite sure if this works this easy. First thing that comes to mind here: Both standard typescript AST-nodes as well as ts-morph nodes are objects with circular dependencies, so you can not print them (what you would need to do if you want to place it as a string in the code of the component file). So we would have to invent our own generic format to save a type with all of its type information (including children etc., basically including all possibilities of the AST). On the other hand, for a proper parser, we would need to implement this in the webpack plugin too.

Generally, i think you are right that the webpack-plugin should not be opinionated about its output format.
Maybe we can start internally in next-client with the current version of the plugin and think about this problem in the coming days/weeks (would be probably something worth to dicuss about again)